### PR TITLE
Correctly set `ansible_host_key_checking` option for network_cli

### PR DESCRIPTION
**Pull request by NilashishC**
Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

##### SUMMARY
- Fixes https://github.com/ansible/ansible/issues/49254
- The `ansible_host_key_checking` option was not being correctly read when set through a var.
- Adding the options in network_cli and setting this on the connection object fixes this issue.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
network_cli.py

### CODE REVIEW

**Review by Qalthos — APPROVED** [review]

File: plugins/connection/network_cli.py [hunk 1]
```diff
@@ -269,6 +269,21 @@
...
+    - name: ANSIBLE_HOST_KEY_CHECKING
+    - name: ANSIBLE_SSH_HOST_KEY_CHECKING
+    ini:
+    - section: defaults
+      key: host_key_checking
+    - section: persistent_connection
+      key: host_key_checking
+    vars:
+    - name: ansible_host_key_checking
+    - name: ansible_ssh_host_key_checking
```
Comment (bcoca) [inline plugins/connection/network_cli.py:286]: don't you need the paramiko ones also?

Comment (ganeshrn) [reply plugins/connection/network_cli.py:286]: People using network_cli connection most likely won't be referring to paramiko plugin options so IMO it is not required. Alternately if we decide to add paramiko options we will also require to add options of libssh plugin.

Comment (NilashishC) [reply plugins/connection/network_cli.py:286]: iiuc, if we add both _ibssh and paramiko options here, the behavior might be a bit misleading, i.e., setting `ansible_paramiko_host_key_checking: False`, would turn of host key checking even when libssh is used underneath, and vice versa.

**Comment by Qalthos** [comment]
recheck

**Pull request merged**